### PR TITLE
Refactor home screen

### PR DESCRIPTION
**Pull request by thornbill**
**Changes**
Refactor home screen code to use separate files per home screen section and remove some unused home screen types

**Issues**
N/A

### CODE REVIEW

File: src/components/homesections/sections/liveTv.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            showAirDateTime: false,
+            showAirEndTime: true,
+            defaultShape: getBackdropShape(enableOverflow),
+            lines: 3,
+            overlayPlayButton: true
+        })
+    );
+}
+
+function buildSection(
```
Comment (ZenenTreadwell) [inline src/components/homesections/sections/liveTv.ts:57]: This seems rather un-reactive. I'm sure there's a better way to do this with JSX.

File: src/components/homesections/sections/nextUp.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    };
+}
+
+export function loadNextUp(
+    elem: HTMLElement,
+    apiClient: ApiClient,
+    userSettings: UserSettings,
+    options: SectionOptions
+) {
+    let html = '';
```
Comment (ZenenTreadwell) [inline src/components/homesections/sections/nextUp.ts:67]: Same here - I don't know if optionally concatenating HTML text is the way to go here when JSX supports conditional rendering etc.

File: src/components/homesections/sections/recentlyAdded.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+function renderLatestSection(
+    elem: HTMLElement,
+    apiClient: ApiClient,
+    user: UserDto,
+    parent: BaseItemDto,
+    options: SectionOptions
+) {
+    let html = '';
```
Comment (ZenenTreadwell) [inline src/components/homesections/sections/recentlyAdded.ts:93]: needs JSX

File: src/components/homesections/sections/recentlyAdded.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    apiClient: ApiClient,
+    user: UserDto,
+    userViews: BaseItemDto[],
+    options: SectionOptions
+) {
+    elem.classList.remove('verticalSection');
+    const excludeViewTypes = ['playlists', 'livetv', 'boxsets', 'channels'];
+    const userExcludeItems = user.Configuration?.LatestItemsExcludes ?? [];
+
+    for (let i = 0, length = userViews.length; i < length; i++) {
```
Comment (ZenenTreadwell) [inline src/components/homesections/sections/recentlyAdded.ts:142]: This seems like a good place to use ES6's userViews.forEach() syntax

File: src/components/homesections/sections/resume.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    titleLabel: string,
+    mediaType: BaseItemKind,
+    userSettings: UserSettings,
+    options: SectionOptions
+) {
+    let html = '';
+
+    const dataMonitor = dataMonitorHints[mediaType] ?? 'markplayed';
+
+    html += '<h2 class="sectionTitle sectionTitle-cards padded-left">' + globalize.translate(titleLabel) + '</h2>';
```
Comment (ZenenTreadwell) [inline src/components/homesections/sections/resume.ts:84]: JSX

**Review by ZenenTreadwell — COMMENTED** [review]
Awesome job doing the refactoring work here! I see a couple places where the code could be improved for readability & conforming to React development standards.

Thanks for putting the effort in!

File: src/components/homesections/sections/liveTv.ts (hunk 1 above)
Comment (grhallenbeck) [reply src/components/homesections/sections/liveTv.ts:57]: The source files these are being split off from are legacy and unfortunately not React. Splitting the legacy pieces out this way should make it easier to divide work on migrating to React going forward, though!

Comment (ZenenTreadwell) [reply src/components/homesections/sections/liveTv.ts:57]: Oops! thanks for saying something

Comment (thornbill) [reply src/components/homesections/sections/liveTv.ts:57]: Thanks @grhallenbeck that is exactly the idea! I just had not made it back to this to comment myself. :sweat_smile:

**Review by grafixeyehero — APPROVED** [review]

**Comment by bradbeattie** [comment]
Regression of https://github.com/jellyfin/jellyfin-web/pull/1134 as a result of this change. Would have expected "Recently Added in Movies" to continue to use the Primary image instead of Backdrops.

image

**Pull request merged**